spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/ProxyRequestHelper.java

```diff
@@ -129,8 +131,14 @@ public void setResponse(int status, InputStream entity,
 			RequestContext.getCurrentContext().setResponseDataStream(entity);
 		}
 		boolean isOriginResponseGzipped = false;
-		if (headers.containsKey(CONTENT_ENCODING)) {
-			Collection<String> collection = headers.get(CONTENT_ENCODING);
+		if (headers.containsKey(CONTENT_ENCODING) || headers.containsKey(CONTENT_ENCODING_LOWER)) {
```

Comment:
Do what `buildZuulRequestHeaders` does: create a new `HttpHeaders` add them, iterate through all of the response headers, and then do your logic know that you have a case insensitive map. It's likely that in the future we'll use other headers besides Content Encoding .